### PR TITLE
Fixes #15558: "Technique diverges" popup is broken

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://issues.rudder.io/issues/15558

### CODE REVIEW

File: builder/js/ncf.js [hunk 1]
```diff
@@ -939,12 +940,12 @@ $scope.onImportFileChange = function (fileEl) {
...
   // Check if a technique has not been changed, and if we can use reset function
   $scope.isUnchanged = function(technique) {
-    return (!$scope.suppressFlag && angular.equals(technique, $scope.originalTechnique));
+    return angular.equals(technique, $scope.originalTechnique);
   };
 
   // Check if a technique has been saved,
   $scope.isNotSaved = function() {
-    return $scope.originalTechnique !== undefined && $scope.originalTechnique.bundle_name === undefined;
+    return $scope.originalTechnique != undefined && $scope.originalTechnique.bundle_name === undefined;
```
Comment (VinceMacBuche) [inline builder/js/ncf.js]: Why removing the second = in !==

File: builder/js/ncf.js [hunk 2]
```diff
@@ -1133,12 +1134,13 @@ $scope.onImportFileChange = function (fileEl) {
   // selected technique is correct if:
   // * There is at least one method call
   $scope.checkSelectedTechnique = function() {
+    if($scope.selectedTechnique===undefined || $scope.selectedTechnique===undefined) return false;
```
Comment (VinceMacBuche) [inline builder/js/ncf.js]: You check the same value twice, remove second $scope.selectedTechnique===undefined

File: builder/js/ncf.js (hunk 1 above)
Comment (RaphaelGauthier) [reply builder/js/ncf.js]: We do not only want to check if is `$scope.originalTechnique` is strictly different from undefined, we want to check if it is different from **null** too. I could write `($scope.originalTechnique !== undefined && $scope.originalTechnique !== undefined)` instead

File: builder/js/ncf.js (hunk 2 above)
Comment (RaphaelGauthier) [reply builder/js/ncf.js]: My bad, I would probably want to check if it is striclty different from null too

**Comment by RaphaelGauthier** [comment]
Commit modified

**Comment by RaphaelGauthier** [comment]
PR replaced by https://github.com/Normation/ncf/pull/1130

**Pull request closed without merging**